py/plugins/google-genai/src/genkit/plugins/google_genai/models/gemini.py

```diff
@@ -538,12 +538,30 @@ def _convert_schema_property(
         Returns:
             Schema or None
         """
-        if input_schema is None or 'type' not in input_schema:
+        if input_schema is None:
             return None
 
         if defs is None:
             defs = input_schema.get('$defs') if '$defs' in input_schema else {}
 
+        if '$ref' in input_schema:
```

Comment:
could you please add a unittest for this fix: https://github.com/firebase/genkit/blob/927dc6aa1f70ccefa54225663f20fbe353cf0a45/py/plugins/google-genai/test/models/test_googlegenai_gemini.py#L193